Code/Tools/AssetProcessor/native/ui/ProductDependencyTreeDelegate.cpp

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include <native/ui/ProductDependencyTreeDelegate.h>
+#include <native/ui/ProductDependencyTreeItemData.h>
+#include <native/ui/ProductAssetDetailsPanel.h>
+#include <native/ui/ProductAssetTreeItemData.h>
+
+#include <QMouseEvent>
+#include <QApplication>
+#include <AzCore/Debug/Trace.h>
+#include <iostream>
+namespace AssetProcessor
+{
+    ProductDependencyTreeDelegate::ProductDependencyTreeDelegate(QObject* parent, QPointer<ProductAssetDetailsPanel> panel)
+        : QStyledItemDelegate(parent)
+        , m_panel(panel)
+    {
+    }
+    bool ProductDependencyTreeDelegate::editorEvent(
+        QEvent* event, QAbstractItemModel* /*model*/, const QStyleOptionViewItem& option, const QModelIndex& index)
```

Comment:
Is this supposed to be [[maybe_used]] QAbstractItemModel* model ?

Reply:
yes. updated. thanks for the catch.

Reply:
The build should error if this is run on AR